packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -229,17 +229,21 @@ void saveStateBeforeResult() {
 
   void retrieveLostImage(MethodChannel.Result result) {
     Map<String, Object> resultMap = cache.getCacheMap();
-    String path = (String) resultMap.get(cache.MAP_KEY_PATH);
-    if (path != null) {
-      Double maxWidth = (Double) resultMap.get(cache.MAP_KEY_MAX_WIDTH);
-      Double maxHeight = (Double) resultMap.get(cache.MAP_KEY_MAX_HEIGHT);
-      int imageQuality =
-          resultMap.get(cache.MAP_KEY_IMAGE_QUALITY) == null
-              ? 100
-              : (int) resultMap.get(cache.MAP_KEY_IMAGE_QUALITY);
-
-      String newPath = imageResizer.resizeImageIfNeeded(path, maxWidth, maxHeight, imageQuality);
-      resultMap.put(cache.MAP_KEY_PATH, newPath);
+    ArrayList<String> pathList = (ArrayList<String>) resultMap.get(cache.MAP_KEY_PATH_LIST);
+    ArrayList<String> newPathList = new ArrayList<>();
+    if (pathList != null) {
+      for (String path : pathList) {
+        Double maxWidth = (Double) resultMap.get(cache.MAP_KEY_MAX_WIDTH);
+        Double maxHeight = (Double) resultMap.get(cache.MAP_KEY_MAX_HEIGHT);
+        int imageQuality =
+            resultMap.get(cache.MAP_KEY_IMAGE_QUALITY) == null
+                ? 100
+                : (int) resultMap.get(cache.MAP_KEY_IMAGE_QUALITY);
+
+        newPathList.add(imageResizer.resizeImageIfNeeded(path, maxWidth, maxHeight, imageQuality));
+      }
+      resultMap.put(cache.MAP_KEY_PATH_LIST, newPathList);
+      resultMap.put(cache.MAP_KEY_PATH, newPathList.get(0));
```

Comment:
I'm not 100% sure on this, but to me it looks like the current behavior is that `file` reflects the last file in the list when multiple images were picked. Setting the 1st file here would probably be a breaking change. I'd suggest changing this to instead set the last path in the list.